metrics/src/main/java/org/aerogear/mobile/metrics/interfaces/MetricsProvider.java

```diff
@@ -0,0 +1,40 @@
+package org.aerogear.mobile.metrics.interfaces;
+
+import android.content.Context;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Objects;
+
+public abstract class MetricsProvider {
+    public String namespace() {
+        return null;
+    }
+
+    public JSONObject metrics(final Context context) throws JSONException {
```

Comment:
does it make sense to have default implementations for these methods? Would it make sense for them to be interfaces instead and force the implementer to implement.

Reply:
+1 for making this abstract (with javadoc documentation)

Reply:
The idea was that the default implementation provides everything that is needed to hook into the metrics service. The `namespace` for example is optional, i would prefer not to enforce implementation in an interface (if you don't override it your properties will be added to the root of the metrics structure). Also i'd like to have default implementations for equals and hashCode that take the namespace into account.

Reply:
@pb82 Yeah. We have abstract class with no abstract method. We could just make this method abstract and that will be all.